### ISSUE TITLE
EMFILE: too many open files

### ISSUE THREAD

**Issue by radum** [GoogleChromeLabs/ndb#27]
### Steps to reproduce

* ndb version: 1.06
* Platform / OS version: macOS High Sierra 10.13.6
* Node.js version: v10.7.0, npm v6.2.0

**What steps will reproduce the problem?**

```shell
npm i -g ndb
ndb test.js
```

_Please include code that reproduces the issue._

```js
function p1() {
	return new Promise((resolve, reject) => {
		setTimeout(() => {
			console.log('p1')
			resolve('p1');
		}, 3000);
	})
}

function p2() {
	return new Promise((resolve, reject) => {
		setTimeout(() => {
			console.log('p2')
			resolve('p2');
		}, 1000);
	})
}
function all() {
	return Promise.all([
		p1(),
		p2()
	]);
}

console.log(all().then((foo) => {
	console.log(foo);
}));
```

**What happens instead?**
```shell
ndb ~/test.js
Debugger listening on ws://127.0.0.1:50703/ac178172-a158-4ef8-bb19-f7127e29ca63
For help, see: https://nodejs.org/en/docs/inspector
Debugger listening on ws://127.0.0.1:50704/85b5cbff-d21b-4c98-bbbf-e42521324d6a
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
Debugger attached.
Promise { <pending> }
p2
2018-07-23 10:54 node[13054] (FSEvents.framework) FSEventStreamStart: register_with_server: ERROR: f2d_register_rpc() => (null) (-22)
2018-07-23 10:54 node[13054] (FSEvents.framework) FSEventStreamStart: register_with_server: ERROR: f2d_register_rpc() => (null) (-22)
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error: EMFILE: too many open files, watch
    at FSEvent.FSWatcher._handle.onchange (internal/fs/watchers.js:120:28)
Emitted 'error' event at:
    at FSEvent.FSWatcher._handle.onchange (internal/fs/watchers.js:126:12)
```

The debugger opens but it closes itself after 2 seconds and the error above fires.

I haven't changed my Mac ulimit or max files. I tried setting

```shell
sudo launchctl limit maxfiles 2048 unlimited
```

from 256 as it is the default, even went up to 6000 but still the same error.

I know in the code files are watched from changes using the Node native API, but there might be better alternatives that will not break the OS limit. In the mean time not sure how to fix this.

**Comment by aberonni** [comment]
I'm having the same issue with 

- ndb v1.0.9 
- macOS High Sierra 10.13.6
- node v10.7.0 (also tested with v8.11.3 with same results)

It was working fine a couple of days ago with a previous version. I didn't change anything and just tried the same command again and it broke, so I tried upgrading `ndb` without success.

**Comment by ak239** [comment]
I just published 1.0.13 that uses [chokidar](https://github.com/paulmillr/chokidar) instead of builtin watchers. This issue should be fixed.
